main.go

```diff
@@ -367,6 +370,16 @@ func main() { // nolint:gocyclo
 		Scheme:         mgr.GetScheme(),
 		Namespace:      namespace,
 		MemberClusters: clusterScopedMemberClusters,
+		GetPublicViewerConfig: func(ctx context.Context, cl client.Client) spacebindingcleanup.PublicViewerConfig {
+			cfg, err := toolchainconfig.GetToolchainConfig(cl)
+			if err != nil {
+				log.FromContext(ctx).
+					Error(err, "error retrieving PublicViewerConfig, using default")
+				return toolchainconfig.PublicViewerConfig{}
+			}
+
+			return cfg.PublicViewer()
+		},
```

Comment:
do we really need to pass this function from here? 

Alternatively you could read the config in the controller reconcile directly using something like:
```
config, err := toolchainconfig.GetToolchainConfig(r.Client)
```
this will avoid all the additional code: 
- changes the main.go
- interfaces and structs definitions in the controller
- mock functions in the tests 

WDYT?

Reply:
done in 022a04807f71b648f7564ca41bc0e9f4e16b9d29